Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -0,0 +1,201 @@
+	
+	var status = isDeploymentSuccess ? 4 : 3;
+	var author = getDeploymentAuthor();
+
+	var buildUrl = tl.getVariable('build.buildUri'); 
+	var	releaseUrl = tl.getVariable ('release.releaseUri');
+
+	var buildId = tl.getVariable('build.buildId');
+	var	releaseId = tl.getVariable('release.releaseId');
+
+	var collectionUrl = tl.getVariable ('system.TeamFoundationCollectionUri'); 
+	var	teamProject = tl.getVariable ('system.teamProject');	
+
+	var buildOrReleaseUrl = "" ;
+	var deploymentId = "";
+
+	if(releaseUrl !== undefined) {
+		deploymentId = releaseId + Date.now();
+		buildOrReleaseUrl = collectionUrl + teamProject + "/_apps/hub/ms.vss-releaseManagement-web.hub-explorer?releaseId=" + releaseId + "&_a=release-summary";
+	}		
+	else if(buildUrl !== undefined){
+		deploymentId = buildId + Date.now();
+		buildOrReleaseUrl = collectionUrl + teamProject + "/_build?buildId=" + buildId + "&_a=summary";
+	}
+	else {
+		throw new Error(tl.loc('CannotupdatedeploymentstatusuniquedeploymentIdCannotBeRetrieved'));
+	}		
+
+	var message = "Updating Deployment History For Deployment " + buildOrReleaseUrl;
+	var requestBody = {
+		status : status, 
+		status_text : status == 4 ? "success" : "failed", 
+		message : message, 
+		author : author, 
+		deployer : 'VSTS', 
+		details : buildOrReleaseUrl
+	};
+	
+	var webAppHostUrl = webAppPublishKuduUrl.split(':')[0];
+	var requestUrl = "https://" + encodeURIComponent(webAppHostUrl) + "/deployments/" + encodeURIComponent(deploymentId);
+
+	var requestDetails = new Array<string>();
+	requestDetails["requestBody"] = requestBody;
+	requestDetails["requestUrl"] = requestUrl;
+	return requestDetails;
+}
+
+async function getWebAppPublishProfile(SPN, webAppName: string, resourceGroupName: string, deployToSlotFlag: boolean, slotName: string ) {
+
+    var deferred = Q.defer();
+    var slotUrl = deployToSlotFlag ? "/slots/" + slotName : "";
+    var accessToken = await getAuthorizationToken(SPN);
+	
+	var requestOptions = {
+		url: armUrl + 'subscriptions/' + SPN.subscriptionId + '/resourceGroups/' + resourceGroupName +
+				 '/providers/Microsoft.Web/sites/' + webAppName + slotUrl + '/publishxml?api-version=2015-08-01',
+		auth: {
+			bearer: accessToken
+		},
+		method: 'POST'
+	};
+	request(requestOptions, (error, response, body) => {
+		if(error) {
+			deferred.reject(error);
+		}
+		else if(response.statusCode === 200) {
+			parseString(body, (error, result) => {
+				for (var index in result.publishData.publishProfile) {
+					if (result.publishData.publishProfile[index].$.publishMethod === "MSDeploy")
+						deferred.resolve(result.publishData.publishProfile[index].$);
+				}
+				deferred.reject(tl.loc('ErrorNoSuchDeployingMethodExists'));
+			});
+		}
+		else {
+			deferred.reject(tl.loc('ErrorFetchingDeploymentPublishProfileStausCode0', response.statusCode));
+		}		
+	});
+
+    return deferred.promise;
+}
+
+async function getAzureRMWebAppDetails(SPN, webAppName: string, resourceType: string) {
```

Comment:
Would be good to add comments at the method level to describe inputs and expected return value